src/DataTable/DataTable.js

```diff
@@ -142,6 +143,17 @@ const DataTable = memo(({
     visibleOnly: selectableRowsVisibleOnly,
     persistSelectedOnPageChange,
   });
+  const handleSortChange = (column, columnDirection) => dispatch({
```

Comment:
Instead of calling sort in a different place than TableCol can't you just pass the `sortResetDefaultField` via context to TableCol and handle the effect there. You can re-use the handleSort logic there